travis/apis.py

```diff
@@ -0,0 +1,174 @@
+            'Accept': 'application/json',
+            'User-Agent': 'mqt',
+            'Authorization': 'Token %s' % self._token
+        })
+        self._request(self.host)
+
+    def _request(self, url, payload=None, is_json=True, patch=False):
+        try:
+            if not payload and not patch:
+                response = self.session.get(url)
+            elif patch:
+                response = self.session.patch(url, data=payload)
+            else:
+                response = self.session.post(url, data=payload)
+            response.raise_for_status()
+        except requests.RequestException as error:
+            raise ApiException(str(error))
+        return response.json() if is_json else response
+
+
+class WeblateApi(Request):
+
+    def __init__(self):
+        super(WeblateApi, self).__init__()
+        self.repo_slug = None
+        self.branch = None
+        self._token = os.environ.get("WEBLATE_TOKEN")
+        self.host = os.environ.get(
+            "WEBLATE_HOST", "https://weblate.vauxoo.com/api")
+        self.tempdir = os.path.join(tempfile.gettempdir(), 'weblate_api')
+
+    def get_project(self, repo_slug, branch):
+        self.repo_slug = repo_slug
+        self.branch = branch
+        projects = self._request(self.host + '/projects')
+        for project in projects['results']:
+            if project['name'] == repo_slug:
+                return project
+        raise ApiException('No project found in "%s" for this path "%s"' %
+                           (self.host, repo_slug))
+
+    def load_project(self, repo_slug, branch):
+        self.project = self.get_project(repo_slug, branch)
+        self.load_components()
+
+    def get_components(self):
+        components = []
+        values = self._request(
+            self.host + '/projects/%s/components/' % self.project['slug'])
+        if not values['results']:
+            raise ApiException('No components found in the project "%s"' %
+                               self.project['slug'])
+        for value in values['results']:
+            if value['branch'] and value['branch'] != self.branch:
+                continue
+            components.append(value)
+        return components
+
+    def load_components(self):
+        self.components = self.get_components()
+
+    def pull(self):
+        pull = self._request(
+            self.host + '/projects/%s/repository/' % self.project['slug'],
+            {'operation': 'pull'})
+        return pull['result']
+
+    def component_repository(self, component, operation):
+        result = self._request(self.host + '/components/%s/%s/repository/' %
+                               (self.project['slug'], component['slug']),
+                               {'operation': operation})
+        return result['result']
+
+    @contextmanager
+    def componet_lock(self):
+        try:
+            for component in self.components:
+                self._component_toggle(component)
```

Comment:
@moylop260 
Done I made the method `_component_lock(self, component, lock=True)` with the lock parameter that the  default value is True